app/helpers/katello/hosts_and_hostgroups_helper.rb

```diff
@@ -98,5 +115,37 @@ def content_views_for_host(host, options)
       view_options.insert(0, include_blank) if include_blank
       view_options.html_safe
     end
+
+    def kickstart_repository_options(host)
+      view_options = []
+
+      if host.present? && host.operatingsystem.is_a?(Redhat)
+        new_host = ::Host.new
+        new_host.operatingsystem = host.operatingsystem
+        new_host.architecture = host.architecture
+        new_host.content_facet = Host::ContentFacet.new(:lifecycle_environment_id => host.lifecycle_environment_id,
+                                                        :content_view_id => host.content_view_id)
+        new_host.content_source = host.content_source
+        view_options = new_host.operatingsystem.kickstart_repos(new_host).map { |repo| OpenStruct.new(repo) }
+      elsif ["operatingsystem_id", 'content_view_id',
+             'lifecycle_environment_id', 'content_source_id',
+             'architecture_id'].all? { |key| !params[key].blank? }
```

Comment:
thats fine, i'm just saying the 3 line elsif with a block is a bit ugly, can you move that to the start of the method?  Sure it'll be executed needlessly in some cases but this not exactly a compute intensive check